microservice-transaction-sample/README.md

```diff
@@ -65,8 +65,8 @@ The sample application supports the following types of transactions:
 ## Prerequisites
 
 - One of the following Java Development Kits (JDKs):
-  - [Oracle JDK](https://www.oracle.com/java/technologies/downloads/) LTS version (8, 11, or 17)
-  - [OpenJDK](https://openjdk.org/install/) LTS version (8, 11, or 17)
+  - [Oracle JDK](https://www.oracle.com/java/technologies/downloads/) 8
+  - [OpenJDK](https://openjdk.org/install/) 8
```

Comment:
In this samples, we build sample applications (`order service` and `customer service`) from the source code instead of using any binaries or container images. And, it depends on a bit old annotations of Java.

So, if we use Java 11 or Java 17 to build sample applications, we faced the following error.

```console
$ java -version
openjdk version "17.0.9" 2023-10-17
OpenJDK Runtime Environment (build 17.0.9+9-Ubuntu-122.04)
OpenJDK 64-Bit Server VM (build 17.0.9+9-Ubuntu-122.04, mixed mode, sharing)
```
```console
$ ./gradlew docker

> Task :rpc:compileJava FAILED
/home/ubuntu/tmp/scalardb-samples/microservice-transaction-sample/rpc/src/main/grpc/sample/rpc/CustomerServiceGrpc.java:10: error: cannot find symbol
@javax.annotation.Generated(
                 ^
  symbol:   class Generated
  location: package javax.annotation
/home/ubuntu/tmp/scalardb-samples/microservice-transaction-sample/rpc/src/main/grpc/sample/rpc/OrderServiceGrpc.java:10: error: cannot find symbol
@javax.annotation.Generated(
                 ^
  symbol:   class Generated
  location: package javax.annotation
2 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':rpc:compileJava'.
> Compilation failed; see the compiler error output for details.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.6/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 871ms
4 actionable tasks: 1 executed, 3 up-to-date
```

To avoid this error, we have to use Java 8. I think we should update sample application to support Java 11 or later.

Reply:
Adding the following dependency might help. Can you please try this?
```
    implementation "javax.annotation:javax.annotation-api:1.3.2"
```

Reply:
I see. I will try it! Thank you for your suggestion! 🙇‍♀️

Reply:
@brfrn169  
I tried to add `implementation "javax.annotation:javax.annotation-api:1.3.2"` in `build.gradle` of customer service and order service.
However, unfortunately, I faced the same error as follows...

```console
$ java -version
openjdk version "17.0.9" 2023-10-17
OpenJDK Runtime Environment (build 17.0.9+9-Ubuntu-122.04)
OpenJDK 64-Bit Server VM (build 17.0.9+9-Ubuntu-122.04, mixed mode, sharing)
```

```diff
$ git diff
diff --git a/microservice-transaction-sample/customer-service/build.gradle b/microservice-transaction-sample/customer-service/build.gradle
index 06cef10..605c7ca 100644
--- a/microservice-transaction-sample/customer-service/build.gradle
+++ b/microservice-transaction-sample/customer-service/build.gradle
@@ -12,6 +12,7 @@ dependencies {
     implementation "org.apache.logging.log4j:log4j-api:${log4jVersion}"
     implementation "org.apache.logging.log4j:log4j-core:${log4jVersion}"
     implementation "org.apache.logging.log4j:log4j-slf4j-impl:${log4jVersion}"
+    implementation "javax.annotation:javax.annotation-api:1.3.2"
 }

 application {
diff --git a/microservice-transaction-sample/order-service/build.gradle b/microservice-transaction-sample/order-service/build.gradle
index 6a9bbf3..b689c0f 100644
--- a/microservice-transaction-sample/order-service/build.gradle
+++ b/microservice-transaction-sample/order-service/build.gradle
@@ -12,6 +12,7 @@ dependencies {
     implementation "org.apache.logging.log4j:log4j-api:${log4jVersion}"
     implementation "org.apache.logging.log4j:log4j-core:${log4jVersion}"
     implementation "org.apache.logging.log4j:log4j-slf4j-impl:${log4jVersion}"
+    implementation "javax.annotation:javax.annotation-api:1.3.2"
 }

 application {
```

```console
$ ./gradlew docker

> Task :rpc:compileJava FAILED
/home/ubuntu/tmp/scalardb-samples/microservice-transaction-sample/rpc/src/main/grpc/sample/rpc/CustomerServiceGrpc.java:10: error: cannot find symbol
@javax.annotation.Generated(
                 ^
  symbol:   class Generated
  location: package javax.annotation
/home/ubuntu/tmp/scalardb-samples/microservice-transaction-sample/rpc/src/main/grpc/sample/rpc/OrderServiceGrpc.java:10: error: cannot find symbol
@javax.annotation.Generated(
                 ^
  symbol:   class Generated
  location: package javax.annotation
2 errors
```

Reply:
It looks like the compile error occurs in the `rpc` subproject. Can you please try to add the dependency to `rpc/build.gradle`?

Reply:
Ah, sorry. I misunderstood...
I will try it again!

Reply:
After adding `implementation "javax.annotation:javax.annotation-api:1.3.2"` to the `rpc/build.gradle`, I was able to build and run the sample application as follows! Thank you for your suggestion!
So, I will fix it in this PR and revert the update of supported version.

## Build

```diff
$ git diff
diff --git a/microservice-transaction-sample/rpc/build.gradle b/microservice-transaction-sample/rpc/build.gradle
index 2b0a5c0..c8f3a3d 100644
--- a/microservice-transaction-sample/rpc/build.gradle
+++ b/microservice-transaction-sample/rpc/build.gradle
@@ -9,6 +9,7 @@ dependencies {
     api "io.grpc:grpc-protobuf:${grpcVersion}"
     api "io.grpc:grpc-stub:${grpcVersion}"
     api "io.grpc:grpc-services:${grpcVersion}"
+    implementation "javax.annotation:javax.annotation-api:1.3.2"
 }

 protobuf {
```
```console
$ java -version
openjdk version "17.0.9" 2023-10-17
OpenJDK Runtime Environment (build 17.0.9+9-Ubuntu-122.04)
OpenJDK 64-Bit Server VM (build 17.0.9+9-Ubuntu-122.04, mixed mode, sharing)
```
```console
$ ./gradlew docker

> Task :customer-service:docker
#0 building with "default" instance using docker driver

#1 [internal] load build definition from Dockerfile
#1 transferring dockerfile: 312B 0.0s done
#1 DONE 0.1s

...

#9 exporting to image
#9 exporting layers
#9 exporting layers 0.5s done
#9 writing image sha256:3fabd0367a9048de3fd88be601857147d3f4cdb0c220ab7130e08a57a9d8528d done
#9 naming to docker.io/library/sample-order-service:1.0 0.0s done
#9 DONE 0.5s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.6/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 18s
22 actionable tasks: 19 executed, 3 up-to-date
```

## Run

```console
$ ./gradlew :client:run --args="GetCustomerInfo 1"

> Task :client:run
{"id": 1,"name": "Yamada Taro","credit_limit": 10000}
```
```console
$ ./gradlew :client:run --args="PlaceOrder 1 1:3,2:2"

> Task :client:run
{"order_id": "e32e3c64-86c3-43b5-bcc7-dce929dca611"}
```
```console
$ ./gradlew :client:run --args="GetOrder e32e3c64-86c3-43b5-bcc7-dce929dca611"

> Task :client:run
{"order": {"order_id": "e32e3c64-86c3-43b5-bcc7-dce929dca611","timestamp": 1703727222165,"customer_id": 1,"customer_name": "Yamada Taro","statement": [{"item_id": 1,"item_name": "Apple","price": 1000,"count": 3,"total": 3000},{"item_id": 2,"item_name": "Orange","price": 2000,"count": 2,"total": 4000}],"total": 7000}}
```

Reply:
Fixed in 94bedfd65ef88ba6eb837aa8559b3f333164ebda .